ciao-controller/compute.go

```diff
@@ -906,7 +911,16 @@ func createServer(w http.ResponseWriter, r *http.Request, context *controller) {
 	}
 	servers.TotalServers = len(instances)
 
-	b, err := json.Marshal(servers)
+	// set machine ID for OpenStack compatibility
+	server.Server.ID = instances[0].ID
```

Comment:
I'm not sure it is.  The code you link to checks to see whether nInstances is 0 and if it is it returns an error.  But if nInstances was -1, you would get (nil, nil) returned from startWorkload and you would get your panic.  And looking at the code, nInstances could be -1, as is initialised from server.Server.MaxInstances, which comes in via HTTP and so could be anything.